src/apps/dashboard/routes/logs/file.tsx

```diff
@@ -0,0 +1,112 @@
+        isError: error,
+        isPending: loading,
+        data: log,
+        refetch
+    } = useServerLog(fileName ?? '');
+
+    const retry = useCallback(() => refetch(), [refetch]);
+
+    const copyToClipboard = useCallback(async () => {
+        if ('clipboard' in navigator && log) {
+            await navigator.clipboard.writeText(log);
+            toast({ text: globalize.translate('CopyLogSuccess') });
+        }
+    }, [log]);
+
+    const downloadFile = useCallback(() => {
+        if ('URL' in globalThis && log && fileName) {
+            const blob = new Blob([log], { type: 'text/plain' });
+            const url = URL.createObjectURL(blob);
+            const a = document.createElement('a');
+            a.href = url;
+            a.download = fileName;
+            a.click();
+            URL.revokeObjectURL(url);
+        }
+    }, [log, fileName]);
+
+    return (
+        <Page
+            id='logPage'
+            title={fileName}
+            className='mainAnimatedPage type-interior'
+        >
+            <Container className='content-primary'>
+                <Box>
+                    <Typography variant='h1'>{fileName}</Typography>
+
+                    {error && (
+                        <Alert
+                            key='error'
+                            severity='error'
+                            sx={{ mt: 2 }}
+                            action={
+                                <Button
+                                    color='inherit'
+                                    size='small'
+                                    onClick={retry}
+                                >
+                                    {globalize.translate('Retry')}
+                                </Button>
+                            }
+                        >
+                            {globalize.translate('LogLoadFailure')}
+                        </Alert>
+                    )}
+
+                    {loading && <Loading />}
+
+                    {!error && !loading && (
+                        <>
+                            <ButtonGroup variant='contained' sx={{ mt: 2 }}>
+                                <Button
+                                    disabled={!('clipboard' in navigator)}
+                                    startIcon={<ContentCopy />}
+                                    onClick={copyToClipboard}
+                                >
+                                    {globalize.translate('Copy')}
+                                </Button>
+                                <Button
+                                    startIcon={<FileDownload />}
+                                    onClick={downloadFile}
+                                >
+                                    {globalize.translate('Download')}
+                                </Button>
+                            </ButtonGroup>
+
+                            <Card sx={{ mt: 2 }}>
+                                <CardContent>
```

Comment:
Maybe just use `<Paper>` instead of a card?

Reply:
I did switch it to Paper but had to manually add the padding to it. I also noticed I forgot to add `overflow:auto` for large lines so added that now as well.